### PR TITLE
Add sending new boundary corrections to ComputeTimeDerivative

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

Exactly as the title says... Surprisingly short, I suspect.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/DiscontinuousGalerkin/Actions/ComputeTimeDerivative.hpp [hunk 1]
```diff
@@ -1026,58 +1029,111 @@ template <typename ParallelComponent, typename DbTagsList>
...
+               "The current time step id of the volume is "
+                   << time_step_id
+                   << "but the time step id on the mortar with mortar id "
+                   << mortar_id << " is "
+                   << all_mortar_data.at(mortar_id).time_step_id());
+
+        // Reorient the data to the neighbor orientation if necessary
+        if (not orientation.is_aligned()) {
+          ERROR("Currently don't support unaligned meshes");
+          // remote_boundary_data_on_mortar.orient_on_slice(
```
Comment (kidder) [inline src/Evolution/DiscontinuousGalerkin/Actions/ComputeTimeDerivative.hpp]: what still needs to be done to support this?

Comment (nilsdeppe) [reply src/Evolution/DiscontinuousGalerkin/Actions/ComputeTimeDerivative.hpp]: I think just more tests, but I wanted to delay that so I can make sure more of the design works. Same with why LTS isn't immediately added to ApplyBoundaryCorrections (though in that case it's also to keep the PR from being even longer). I also wanted to think about memory allocations a bit more here.

File: src/Evolution/DiscontinuousGalerkin/Actions/ComputeTimeDerivative.hpp [hunk 2]
```diff
@@ -1026,58 +1029,111 @@ template <typename ParallelComponent, typename DbTagsList>
...
+
+    for (const auto& [direction, neighbors] : element.neighbors()) {
+      const auto& orientation = neighbors.orientation();
+      const auto direction_from_neighbor = orientation(direction.opposite());
+
+      for (const auto& neighbor : neighbors) {
+        const std::pair mortar_id{direction, neighbor};
+
+        std::pair<Mesh<volume_dim - 1>, std::vector<double>>
+            remote_boundary_data_on_mortar =
```
Comment (fmahebert) [inline src/Evolution/DiscontinuousGalerkin/Actions/ComputeTimeDerivative.hpp]: For consistency with an earlier PR in this series, should "remote" -> "neighbor" ?

File: src/Evolution/DiscontinuousGalerkin/Actions/ComputeTimeDerivative.hpp [hunk 3]
```diff
@@ -1026,58 +1029,111 @@ template <typename ParallelComponent, typename DbTagsList>
...
+        std::tuple<Mesh<volume_dim - 1>, std::optional<std::vector<double>>,
+                   std::optional<std::vector<double>>, ::TimeStepId>
+            data{remote_boundary_data_on_mortar.first,
+                 {},
+                 {std::move(remote_boundary_data_on_mortar.second)},
+                 db::get<tmpl::conditional_t<Metavariables::local_time_stepping,
+                                             ::Tags::Next<::Tags::TimeStepId>,
+                                             ::Tags::TimeStepId>>(box)};
+
+        // Send remote data to neighbor
```
Comment (fmahebert) [inline src/Evolution/DiscontinuousGalerkin/Actions/ComputeTimeDerivative.hpp]: It's not at all clear to me what this means, though perhaps it's related to my other comment. Does "remote" mean "neighbor" here, as it did in a previous PR? Please clarify...

**Comment by nilsdeppe** [comment]
Rebased on develop, rename `remote` -> `neighbor` (thanks for keeping me consistent!). I squashed since this is a fairly easy change (I think)

File: src/Evolution/DiscontinuousGalerkin/Actions/ComputeTimeDerivative.hpp [hunk 4]
```diff
@@ -1026,58 +1029,111 @@ template <typename ParallelComponent, typename DbTagsList>
...
+        std::tuple<Mesh<volume_dim - 1>, std::optional<std::vector<double>>,
+                   std::optional<std::vector<double>>, ::TimeStepId>
+            data{neighbor_boundary_data_on_mortar.first,
+                 {},
+                 {std::move(neighbor_boundary_data_on_mortar.second)},
+                 db::get<tmpl::conditional_t<Metavariables::local_time_stepping,
+                                             ::Tags::Next<::Tags::TimeStepId>,
+                                             ::Tags::TimeStepId>>(box)};
+
+        // Send neighbor data to neighbor
```
Comment (fmahebert) [inline src/Evolution/DiscontinuousGalerkin/Actions/ComputeTimeDerivative.hpp]: I'd still like to better understand this. Can you (in Github for now) expand a bit on why you're doing this? What is actually being sent (i.e., what is it that you are calling "neighbor data?")?

Comment (nilsdeppe) [reply src/Evolution/DiscontinuousGalerkin/Actions/ComputeTimeDerivative.hpp]: This sends the boundary correction data from self to my neighbor. That is, the result of `dg_package_data` is sent to the neighbor. Suggestions on how to make this clear than `Send data for X to X`? :P

Comment (fmahebert) [reply src/Evolution/DiscontinuousGalerkin/Actions/ComputeTimeDerivative.hpp]: Right, I had interpreted "neighbor data" as being "data from the neighbor" so this seemed like sending the data back where it came from... and surprising :P. What do you think of this phrasing: "Send mortar data to neighbor" ? (If you like it, please squash immediately)

Comment (fmahebert) [reply src/Evolution/DiscontinuousGalerkin/Actions/ComputeTimeDerivative.hpp]: Or "Send flux data to neighbor" ?

(and please make same name change in `else` branch below)

Comment (nilsdeppe) [reply src/Evolution/DiscontinuousGalerkin/Actions/ComputeTimeDerivative.hpp]: Yea, I like that. Add a parenthetical `(the std::tuple named data)` to be a bit extra verbose :) Thanks!

**Review by fmahebert — DISMISSED** [review]

**Review by kidder — DISMISSED** [review]

**Comment by nilsdeppe** [comment]
Rebased on develop to get the CI-related fix

**Review by fmahebert — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Pull request merged**